pkg/utils/release/controller.go

```diff
@@ -424,3 +450,132 @@ func (s *SuiteController) CreateReleasePipelineRoleBindingForServiceAccount(name
+
+			if len(releaseName) > 0 {
+				prList, err = s.GetPipelineRunListInNamespace(managedNamespace, releaseName, devNamespace)
+				if err != nil {
+					GinkgoWriter.Println("PipelineRun has not been created yet.")
+					return false, nil
+				}
+			} else {
+				prList, err = s.GetAllReleasePipelineRunMatchComponent(managedNamespace, devNamespace, component)
+				if err != nil {
+					GinkgoWriter.Println("PipelineRun has not been created yet.")
+					return false, nil
+				}
+
+			}
+
+			pr := &prList.Items[0]
+			GinkgoWriter.Printf("PipelineRun %s reason: %s\n", pr.Name, pr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).GetReason())
+
+			if !pr.IsDone() {
+				return false, nil
+			}
+
+			if pr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsTrue() {
+				return true, nil
+			} else {
+				var prLogs string
+				if err = tekton.StorePipelineRun(pr, s.KubeRest(), s.KubeInterface()); err != nil {
+					GinkgoWriter.Printf("failed to store PipelineRun %s:%s: %s\n", pr.GetNamespace(), pr.GetName(), err.Error())
+				}
+				if prLogs, err = tekton.GetFailedPipelineRunLogs(s.KubeRest(), s.KubeInterface(), pr); err != nil {
+					GinkgoWriter.Printf("failed to get logs for PipelineRun %s:%s: %s\n", pr.GetNamespace(), pr.GetName(), err.Error())
+				}
+				return false, fmt.Errorf(prLogs)
+			}
+		})
+
+		if err != nil {
+			GinkgoWriter.Printf("attempt %d/%d: PipelineRun %q failed: %+v", attempts, maxRetries+1, pr.GetName(), err)
+			attempts++
+		} else {
+			break
+		}
+	}
+
+	return nil
+
+}
+
+// GetAllReleasePipelineRunMatchComponent returns the release pipelineRuns in namespace associated with component.
+func (s *SuiteController) GetAllReleasePipelineRunMatchComponent(managedNamespace, devNamespace string, component *appstudioApi.Component) (*v1beta1.PipelineRunList, error) {
+
+	appName := component.Spec.Application
+
+	snapshot, err := s.GetSnapshotByComponent(devNamespace, component.Name)
+	if err != nil && !k8sErrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	release, err := s.GetReleaseMatchSnapshot(devNamespace, snapshot.Name)
+	if err != nil && !k8sErrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	releasePipelineRunList, err := s.GetReleasePipelineRunByAppNameAndReleaseName(managedNamespace, appName, release.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	return releasePipelineRunList, nil
+}
+
+// GetReleaseMatchSnapshot returns a release in namespace with the given snapshot name.
+func (s *SuiteController) GetReleaseMatchSnapshot(namespace, snapshotName string) (*releaseApi.Release, error) {
+	var release *releaseApi.Release
+	releaseList := &releaseApi.ReleaseList{}
+	opts := []client.ListOption{
+		client.InNamespace(namespace),
+	}
+
+	err := s.KubeRest().List(context.TODO(), releaseList, opts...)
+	if err != nil && !k8sErrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	for _, r := range releaseList.Items {
```

Comment:
This should be passed to the List operation. See https://github.com/redhat-appstudio/release-service/blob/main/loader/loader.go#L58 for an example